### PR TITLE
CSI index support and other minor fixes

### PR DESCRIPTION
**Pull request by praveenraj2018**
Update includes:
1) Use of csi index when works with large genomes using `--bam_csi_index`. Tested on Oppossum genome locally and verified.
2) Update in documentation for the usage for csi index
3) Fix issues when using one known variant VCF resource file instead of both.
4) Update nf-core modules to the latest version available
5) Used SplitNCigarReads from nf-core (instead of a local version)
6) Used GATK4/Markduplicates instead of one from the nf-core Picard

## PR checklist

- [x] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
  - [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/rnavar/tree/master/.github/CONTRIBUTING.md)
  - [ ] If necessary, also make a PR on the nf-core/rnavar _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [x] Make sure your code lints (`nf-core lint`).
- [x] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [x] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

File: conf/modules.config [hunk 1]
```diff
@@ -295,26 +307,29 @@ process {
     withName: GATK4_HAPLOTYPECALLER {
         ext.args    = [
             '-dont-use-soft-clipped-bases',
-            params.stand_call_conf ? "--standard-min-confidence-threshold-for-calling $params.stand_call_conf" : ''
+            params.stand_call_conf ? "--standard-min-confidence-threshold-for-calling $params.stand_call_conf" : '',
+            params.bam_csi_index ? "--create-output-variant-index false" : ""
```
Comment (maxulysse) [inline conf/modules.config:311]: <3

File: lib/WorkflowRnavar.groovy [hunk 2]
```diff
@@ -20,8 +20,8 @@ class WorkflowRnavar {
             System.exit(1)
         }
 
-        if ((!params.skip_baserecalibration) && (!params.known_indels || !params.known_indels_tbi)) {
-            log.error "Known variants VCF file or its index is missing!. Use --known_indels and --known_indels_tbi to supply the files."
+        if ((!params.skip_baserecalibration) && (!params.dbsnp && !params.known_indels)) {
+            log.error "Known variants VCF file or its index is missing!. Atleast --dbsnp (and its index) or --known_indels (and its index) is required."
```
Comment (maxulysse) [inline lib/WorkflowRnavar.groovy:24]:
```suggestion
            log.error "Known variants VCF file or its index is missing!. At least --dbsnp (and its index) or --known_indels (and its index) is required."
```

File: conf/modules.config [hunk 3]
```diff
@@ -237,10 +237,22 @@ process {
...
     }
 
+    withName: '.*:SPLITNCIGAR:GATK4_SPLITNCIGAR:GATK4_SPLITNCIGARREADS' {
+        ext.args    =  '--create-output-bam-index false'
+    }
+
+    withName: '.*:SPLITNCIGAR:GATK4_SPLITNCIGAR:SAMTOOLS_INDEX' {
+        ext.args    =  params.bam_csi_index ? '-c' : ''
+    }
+
```
Comment (maxulysse) [inline conf/modules.config:247]:
```suggestion
```

Isn't that case covered by lines 252 to 254 already?

Comment (praveenraj2018) [reply conf/modules.config:247]: No. it is the indexing of the merged BAM.

Comment (maxulysse) [reply conf/modules.config:247]: Do you think a `withName: '.*:SPLITNCIGAR*SAMTOOLS_INDEX' {` could be working?
but we might loose some clarity.
I'm fine as it is.

File: workflows/rnavar.nf [hunk 4]
```diff
@@ -92,8 +93,10 @@ def seq_platform        = params.seq_platform ? params.seq_platform : []
 def seq_center          = params.seq_center ? params.seq_center : []
 
 // Initialize file channels based on params
-dbsnp                   = params.dbsnp             ? Channel.fromPath(params.dbsnp).collect()       : []
-dbsnp_tbi               = params.dbsnp_tbi         ? Channel.fromPath(params.dbsnp_tbi).collect()   : []
+dbsnp                   = params.dbsnp             ? Channel.fromPath(params.dbsnp).collect()               : Channel.empty()
+dbsnp_tbi               = params.dbsnp_tbi         ? Channel.fromPath(params.dbsnp_tbi).collect()           : Channel.empty()
+known_indels            = params.known_indels      ? Channel.fromPath(params.known_indels).collect()        : Channel.empty()
+known_indels_tbi        = params.known_indels_tbi  ? Channel.fromPath(params.known_indels_tbi).collect()    : Channel.empty()
```
Comment (maxulysse) [inline workflows/rnavar.nf:99]: What about
```suggestion
dbsnp                   = params.dbsnp             ? Channel.fromPath(params.dbsnp).collect()               : []
dbsnp_tbi               = params.dbsnp_tbi         ? Channel.fromPath(params.dbsnp_tbi).collect()           : []
known_indels            = params.known_indels      ? Channel.fromPath(params.known_indels).collect()        : []
known_indels_tbi        = params.known_indels_tbi  ? Channel.fromPath(params.known_indels_tbi).collect()    : []
```
That should take care of lines +292 to +297

Comment (praveenraj2018) [reply workflows/rnavar.nf:99]: if we change to [], then the concat() at line +246 to +247 will fail.
Earlier we used the following with [], but resulted in error (java.lang.NullPointerException: Cannot get property 'id' on null object) that Sarah faced when only known_indel file was provided without a dbSNP.

```
known_sites     = Channel.from([params.dbsnp, params.known_indels]).collect()
known_sites_tbi = Channel.from([params.dbsnp_tbi, params.known_indels_tbi]).collect()
```

**Comment by praveenraj2018** [comment]
Any idea why some the tasks are not started but Required?

**Comment by maxulysse** [comment]
> Any idea why some the tasks are not started but Required?

Fixed

**Pull request merged**